src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,50 +1,111 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isValidUrl = (url: string) => {
+    // eslint-disable-next-line max-len
+    const pattern = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)/;
+
+    return pattern.test(url);
+  };
+
+  const availablButton = !title
+    || !isValidUrl(imgUrl)
+    || !isValidUrl(imdbUrl)
+    || !imdbId;
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        setCount(prevCount => prevCount + 1);
+        setTitle('');
+        setDescription('');
+        setImgUrl('');
+        setImdbUrl('');
+        setImdbId('');
+
+        const newMovie = {
+          title,
+          description,
+          imgUrl,
+          imdbUrl,
+          imdbId,
+        };
+
+        onAdd(newMovie);
+      }}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={event => {
+          setTitle(event);
+        }}
```

Comment:
try to rewrite in all TextFields this way 👇

```suggestion
        onChange={setTitle}
```